sisyphus/global_settings.py

```diff
@@ -317,72 +318,53 @@ def file_caching(path):
 #: Raise an exception if a Variable is accessed which is not set yet
 RAISE_VARIABLE_NOT_SET_EXCEPTION = False
 
+# Parameter used for debugging or profiling
+MEMORY_PROFILE_LOG = None
+
+USE_UI = True
+
+# Set to False to disable Warning of unset engine
+ENGINE_NOT_SETUP_WARNING = True
 
 # Internal functions
-def update_global_settings_from_text(text, filename):
-    """
-    :param str text:
-    :param str filename:
-    :return: nothing
-    """
-    # create basically empty globals
-    globals_ = {
-        '__builtins__': globals()['__builtins__'],
-        '__file__': filename,
-        '__name__': filename,
-        '__package__': None,
-        '__doc__': None,
-    }
-    globals_keys = list(globals_.keys())
-
-    # it might be useful to change the default environment by modifying the existing set/dict
-    # thus we need to add it to the globals, but not to globals_keys
-    globals_['DEFAULT_ENVIRONMENT_KEEP'] = DEFAULT_ENVIRONMENT_KEEP
-    globals_['DEFAULT_ENVIRONMENT_SET'] = DEFAULT_ENVIRONMENT_SET
-
-    # compile is needed for a nice trace back
-    exec(compile(text, filename, "exec"), globals_)
-
-    for k, v in globals_.items():
-        if k not in globals_keys:
-            globals()[k] = v
+ENVIRONMENT_SETTINGS = {}
+ENVIRONMENT_SETTINGS_PREFIX = 'SIS_'
 
-    if AUTO_SET_JOB_INIT_ATTRIBUTES:
-        import logging
-        logging.warning('AUTO_SET_JOB_INIT_ATTRIBUTES is deprecated, please set the attributes manually '
-                        'you might want to use self.set_attrs(locals())')
+#: Stores content of all given settings file allowing to log and recreate them if necessary
+GLOBAL_SETTINGS_FILE_CONTENT = ''
 
 
 def update_global_settings_from_file(filename):
-    """
+    """ Loads setting file and updates state of global_settings with its content
+
     :param str filename:
     :return: nothing
     """
-    # skip if settings file doesn't exist
-    globals()['GLOBAL_SETTINGS_FILE'] = filename
+    global GLOBAL_SETTINGS_FILE_CONTENT
 
-    content = ''
     try:
         with open(filename, encoding='utf-8') as f:
             content = f.read()
     except IOError as e:
-        if e.errno != 2:
+        if e.errno == 2:
```

Comment:
You could either just catch `FileNotFoundError`, or better use the `errno` module and check for `errno.ENOENT`.